src/GraphQL/Resolver/Element.php

```diff
@@ -136,6 +148,165 @@ public function resolveSiblings($value = null, $args = [], $context, ResolveInfo
         return [];
     }
 
+    /**
+     * @param null $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return mixed
+     */
+    public function resolveEdge($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        $element = $value['node'];
+
+        if (null !== $element) {
+            return $this->extractSingleElement($element, $args, $context, $resolveInfo);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return mixed
+     */
+    public function resolveEdges($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        return $value['edges'];
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return array
+     * @throws \Exception
+     */
+    public function resolveListing($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        if ($args && isset($args['defaultLanguage'])) {
+            $this->getGraphQlService()->getLocaleService()->setLocale($args['defaultLanguage']);
+        }
+
+        $modelFactory = $this->getGraphQlService()->getModelFactory();
+        $listClass = $this->getListType();
+
+        /** @var AbstractListing $objectList */
+        $objectList = $modelFactory->build($listClass);
+        $conditionParts = [];
+        if (isset($args['ids'])) {
+            $conditionParts[] = '(o_id IN (' . $args['ids'] . '))';
+        }
+
+        // paging
+        if (isset($args['first'])) {
+            $objectList->setLimit($args['first']);
+        }
+
+        if (isset($args['after'])) {
+            $objectList->setOffset($args['after']);
+        }
+
+        // sorting
+        if (!empty($args['sortBy'])) {
+            $objectList->setOrderKey($args['sortBy']);
+            if (!empty($args['sortOrder'])) {
+                $objectList->setOrder($args['sortOrder']);
+            }
+        }
+
+        // Include unpublished
+//        if (isset($args['published']) && $args['published'] === false) {
+//            $objectList->setUnpublished(true);
+//        }
+
+        /** @var $configuration Configuration */
+        $configuration = $context['configuration'];
+//        $sqlListCondition = $configuration->getSql();
+//
+//        if ($sqlListCondition) {
+//            $conditionParts[] = '(' . $sqlListCondition . ')';
+//        }
+
+        // check permissions
+        $db = Db::get();
+        $tableName = $this->getTypeTableName();
+        $workspacesTableName = 'plugin_datahub_workspaces_' . $this->elementType;
+        $conditionParts[] = ' (
```

Comment:
this query doesn't work properly, not sure why, I copied it from objects

Reply:
what is the query and what is the error ?

Reply:
fixed it